Bottom: 39f80103c7ed1e6dd3ba3b39a722d5650e408a13
Top:    1291b15009630faabeebe748eff76befbc3a0236
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-07-11 17:03:11 -0500

Create new role buildbot-worker-assets

Starts a web server for distributing assets to buildbot workers.


---
diff --git a/ansible-playbook/buildbot.yml b/ansible-playbook/buildbot.yml
index 63cebaa2..e7c3a099 100644
--- a/ansible-playbook/buildbot.yml
+++ b/ansible-playbook/buildbot.yml
@@ -26,6 +26,19 @@
       new_user_email: buildbot@novatech-llc.com
       new_user_pass: "{{ lookup('password', '/dev/null') }}"
 
+- name: Setup Buildbot worker system
+  hosts: builder
+  become: true
+  become_user: root
+  become_method: sudo
+  tags:
+  - buildbot
+  - buildbot-worker
+  vars:
+    buildbot_master: "{{ hostvars.buildbot_master.ip_addr }}"
+  roles:
+  - role: buildbot-worker-assets
+
 - name: Gitlab public keys for buildbot containers
   hosts: daedalus
   become: true
diff --git a/roles/buildbot-worker-assets/tasks/main.yml b/roles/buildbot-worker-assets/tasks/main.yml
new file mode 100644
index 00000000..0d698a80
--- /dev/null
+++ b/roles/buildbot-worker-assets/tasks/main.yml
@@ -0,0 +1,16 @@
+---
+- name: Start asset server
+  docker_container:
+    name: buildbot-assets
+    image: "httpd:alpine"
+    volumes:
+    - "/mnt/assets:/usr/local/apache2/htdocs/:ro"
+    ports:
+    - 80
+  register: asset_server
+
+- set_fact:
+    buildbot_asset_host: "http://\
+    {{asset_server.ansible_facts.docker_container.NetworkSettings.Gateway}}\
+    :{{asset_server.ansible_facts.docker_container.NetworkSettings.Ports['80/tcp'][0].HostPort}}\
+    "
